### PR TITLE
fix(CX-1929): home page spacing fix

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Bugfix**

This PR resolves [CX-1929]

### Description

This PR fixes the spacing issues between rails on the home page, specifically when a new user does not have all of the rails yet.  

Note: make a new account to test. 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Spacing issues between home page rails is fixed - rquartararo

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

#### Screen recording

https://user-images.githubusercontent.com/50849237/133589242-1ddc6c02-0afc-4838-8151-e38f8d443372.mp4

 

[CX-1929]: https://artsyproduct.atlassian.net/browse/CX-1929

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Spacing issues between home page rails is fixed - rquartararo)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b5968d5453907eb23e1c97932904182d272458e8
</p>

**Comment by rquartararo** [comment]
There is still an issue with a unique use case in which a user views only one artwork and that particular artwork does't generate the "Similar to Works You've Viewed" rail.

**Comment by rquartararo** [comment]
I think this method may be a good short term solution, but I think we need to eventually reactor the home page to conditionally generate the "rowData" based on if the rails have data and are therefore going to render -- as opposed to having the hide component logic inside each rail after they are already passed into the flatlist. What do you think? @olerichter00 @lordkiz

File: src/lib/Scenes/Home/Components/SalesRail.tsx [hunk 1]
```diff
@@ -39,11 +43,16 @@ const SalesRail: React.FC<Props & RailScrollProps> = (props) => {
     return `${subtitle} ${bullet} ${dateAt}`
   }
 
-  useImperativeHandle(props.scrollRef, () => ({
+  useImperativeHandle(scrollRef, () => ({
     scrollToTop: () => listRef.current?.scrollToOffset({ offset: 0, animated: false }),
   }))
 
-  if (!props.salesModule.results?.length) {
+  // Checks to see if the artworks rail is being rendered and hides the separator accordingly
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Components/SalesRail.tsx:50]: Can we also do this check inside of the artworks rail and hide the separator?

**Comment by lordkiz** [comment]
> I think this method may be a good short term solution, but I think we need to eventually reactor the home page to conditionally generate the "rowData" based on if the rails have data and are therefore going to render -- as opposed to having the hide component logic inside each rail after they are already passed into the flatlist. I assume with the new rails being added we will have even more issues with spacing. What do you think? @olerichter00 @lordkiz

I think we can even keep this logic if you don't use onHide/onShow as the prop name. There is no harm passing an explicit prop named hideSeparator for the purpose of hiding separators. I think onHide/onShow sounds like something I can use to chain multiple functions/effects reacting to a show or hide event.

**Comment by olerichter00** [comment]
We probably need this logic in every rail that is personalized. Would be great to find a better solution eventually.

> I think we can even keep this logic if you don't use onHide/onShow as the prop name. There is no harm passing an explicit prop named hideSeparator for the purpose of hiding separators. I think onHide/onShow sounds like something I can use to chain multiple functions/effects reacting to a show or hide event.

The reason why I named it onHide/onShow and not hideSeparator is to decouple the components. The module shouldn't know anything about the separators (it could also be placed in a grid or somewhere else).

**Review by MounirDhahri — APPROVED** [review]
This LGTM. I do believe though that this might be no longer the best way to solve this. A better approach would probably be to use the same pattern as in [Sale.tsx](https://github.com/artsy/eigen/blob/1fdfc9f2c85a41cdbfc7dbf754378f048b23b3ef/src/lib/Scenes/Sale/Sale.tsx#L145-L145) and by leveraging `compact`, we only send the valid rails to the Flatlist and no longer worry about if it's there or not and remove the extra unneeded [coupling](https://proxy.c2.com/cgi/wiki?LawOfDemeterAndCoupling) as per  the [law of demeter](https://proxy.c2.com/cgi/wiki?LawOfDemeter)

**Pull request merged**